src/Federation/EntityResolverProvider.php

```diff
@@ -198,12 +208,13 @@ static function (string $classCandidate): bool {
      * @param  \Illuminate\Support\Collection<\GraphQL\Language\AST\SelectionSetNode>  $keyFieldsSelections
      * @param  array<string, mixed>  $representation
      */
-    protected function constrainKeys(EloquentBuilder $builder, Collection $keyFieldsSelections, array $representation): void
+    protected function constrainKeys(EloquentBuilder $builder, Collection $keyFieldsSelections, array $representation, ObjectTypeDefinitionNode $definition): void
```

Comment:
Could `$definition` also be an `InterfaceTypeDefinitionNode`?

Reply:
It is coming from the method `\Nuwave\Lighthouse\Federation\EntityResolverProvider::typeDefinition` which says it is instance of `ObjectTypeDefinitionNode`, so I guess not 🤔